modules/ceph/src/main/java/org/testcontainers/containers/CephContainer.java

```diff
@@ -0,0 +1,117 @@
+package org.testcontainers.containers;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+
+import java.time.Duration;
+
+@Slf4j
+@Getter
+public class CephContainer extends GenericContainer<CephContainer> {
+
+    public static final String IMAGE = "ceph/daemon";
+    public static final String DEFAULT_TAG = "v3.2.13-stable-3.2-mimic-centos-7";
+
+    public static final int S3_PORT = 8080;
+    public static final int REST_API_PORT = 5000;
+
+    private String awsAccessKey = "ceph";
+
+    private String awsSecretKey = "ceph";
+
+    private String bucketName = "CEPH";
+
+    private String rgwName = "localhost";
+
+    private String demoUid = "ceph";
+
+    private NetworkAutoDetectMode networkAutoDetectMode = NetworkAutoDetectMode.IPV4_ONLY;
+
+    public CephContainer() {
+        super(IMAGE + ":" + DEFAULT_TAG);
+    }
+
+    public CephContainer(String dockerImageName) {
+        super(dockerImageName);
+    }
+
+    @Override
+    protected void configure() {
+        withEnv("RGW_NAME", rgwName);
+        withEnv("NETWORK_AUTO_DETECT", networkAutoDetectMode.value);
+        withEnv("CEPH_DAEMON", "demo");
+        withEnv("CEPH_DEMO_UID", demoUid);
+        withEnv("CEPH_DEMO_ACCESS_KEY", awsAccessKey);
+        withEnv("CEPH_DEMO_SECRET_KEY", awsSecretKey);
+        withEnv("CEPH_DEMO_BUCKET", bucketName);
+        withExposedPorts(REST_API_PORT, S3_PORT);
+        waitingFor(
+                new HttpWaitStrategy()
+                        .forPath("/")
+                        .forPort(REST_API_PORT)
+                        .forStatusCode(200)
+                        .withStartupTimeout(Duration.ofMinutes(5))
+        );
+        withLogConsumer(new Slf4jLogConsumer(log));
+    }
+
+    public AWSCredentialsProvider getAWSCredentialsProvider() {
+        return new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(awsAccessKey, awsSecretKey)
+        );
+    }
+
+    public AwsClientBuilder.EndpointConfiguration getAWSEndpointConfiguration() {
```

Comment:
IIRC we agreed not to use 3rd party types in our public API (plus, I believe it may not work with AWS SDK v2)